torchgeo/datasets/south_africa_crop_type.py

```diff
@@ -0,0 +1,248 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+"""South Africa Crop Type Competition Dataset."""
+
+import os
+import re
+from typing import Any, Callable, Optional, cast
+
+import matplotlib.pyplot as plt
+import torch
+from matplotlib.figure import Figure
+from rasterio.crs import CRS
+from torch import Tensor
+
+from .geo import RasterDataset
+from .utils import BoundingBox, RGBBandsMissingError
+
+
+class SouthAfricaCropType(RasterDataset):
+    """South Africa Crop Type Challenge dataset.
+
+    The `South Africa Crop Type Challenge
+    <https://beta.source.coop/repositories/radiantearth/south-africa-crops-competition/description/>`__
+    dataset includes satellite imagery from Sentinel-1 and Sentinel-2 and labels for
+    crop type that were collected by aerial
+    and vehicle survey from May 2017 to March 2018. Data was collected by the
+    provided by the Western Cape Department of Agriculture and is
+    is available via the Radiant Earth Foundation.
+    Each chip is matched with a label.
+    Each pixel in the label contains an integer field number and crop type class.
+
+    Dataset format:
+
+    * images are 2-band Sentinel 1 and 12-bands Sentinel-2 data with a cloud mask
+    * masks are tiff image with unique values representing the class and field id.
+
+    Dataset classes:
+
+    0: No Data
+    1: Lucerne/Medics
+    2: Planted pastures (perennial)
+    3: Fallow
+    4: Wine grapes
+    5: Weeds
+    6: Small grain grazing
+    7: Wheat
+    8: Canola
+    9: Rooibos
```

Comment:
```suggestion
    0. No Data
    1. Lucerne/Medics
    2. Planted pastures (perennial)
    3. Fallow
    4. Wine grapes
    5. Weeds
    6. Small grain grazing
    7. Wheat
    8. Canola
    9. Rooibos
```
Needed for correct list formatting by Sphinx